bundles/automation/org.eclipse.smarthome.automation.module.script.defaultscope.test/src/test/java/org/eclipse/smarthome/automation/module/script/defaultscope/internal/ScopeTest.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.automation.module.script.defaultscope.internal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.eclipse.smarthome.automation.module.script.ScriptEngineContainer;
+import org.eclipse.smarthome.automation.module.script.ScriptEngineManager;
+import org.eclipse.smarthome.test.java.JavaOSGiTest;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * This tests the script modules
+ *
+ * @author Kai Kreuzer - initial contribution
+ *
+ */
+public class ScopeTest extends JavaOSGiTest {
+
+    private ScriptEngine engine;
+
+    private final String path = "ESH-INF" + File.separator + "automation" + File.separator + "jsr223";
```

Comment:
I'm wondering if there is a more elegant way of loading resources from ESH-INF. @sjka can you give a hint here?

Reply:
Like e.g. `bundleContext.getBundle().getEntry(...)`? Or what do you have in mind?

Reply:
dont know, some OSGi magic I'm not aware of.
`bundleContext.getBundle().getResource(name)` sounds interesting as the class loader would search for `name` in the bundle. Apparently not for fragments what we have here...

Reply:
Good point - in fact `getEntry(...)` does NOT use the class loader, so this indeed might go wrong with fragments. `getResource(...)` would pretty much do the same, using the class loader. That should work as the fragment just extends it's host bundle's class loader.